### DISCUSSION TITLE
Krux firmware dependencies

### DISCUSSION THREAD

**Discussion by nsvrn** [selfcustody/krux#435] [General]
I was looking at the `poetry.lock` file of this repo and was very curious to understand why there are so many dependencies for this project? I even saw a dependency for jupyter/ipython. I would like to understand why a hardware wallet firmware needs all of this or perhaps I'm overlooking something.

**Comment by odudex** [comment]
Don't worry. None of these dependencies are embedded in the firmware.
The project has many other Python scripts and automation tools that require them for tasks such as:
- Code quality inspection and testing
- Translation aids and verification
- Automation tools for documentation and even signing releases

There's also the simulator, which uses some Python packages to mock the dedicated device's hardware when running on a PC.

**Comment by nsvrn** [comment]
Great, that's good to know. Is it possible and easy for a user to compile binaries from the code?  Also, what's the best way to know exact dependencies used for the firmware part?

**Reply by odudex** [reply]
Here's a step-by-step guide to building the firmware.
https://youtu.be/Vmr_TFy2TfQ?si=4KodnpCZ0ORtPHgo
We do everything possible to help users build it. You can join our Telegram group for quick help too.
You don't need any dependencies to build it, not even to install and use Poetry. You'll only need it if you engage in customizing it, so you'll benefit from having all those tools I mentioned above.

**Reply by nsvrn** [reply]
Great, I appreciate the answer.
